### PR TITLE
Hv 562 Cancel and prevent default on Touch contextmenu events

### PR DESCRIPTION
**Pull request by tomconnell-wf**
# Problem
Holds on Windows 8 send a contextmenu(right click) event after release.  Holds also show a black outlined box during the hold, to indicate that a contextmenu is going to appear.  Neither of these events are consistent with the experience in our app.  

# Solution
By default, uicomponents is going to snuff both of these events.  Consumers may override this behavior by passing in options to the EventSynthesizer.

# How to Test   
1. Run ```grunt serve```.
1. Go to the AwesomeMap demo.
1. On a Windows 8 device with a touchscreen, hold on the map.  You should see a black outlined box when holding, and when you release, you should see your browser's context menu open.
1. Switch to this branch.  
1. Run the demo again, and hold on the map.  You should no longer see the box, and when you release, no context menu should appear.

### CODE REVIEW

**Comment by rmconsole-wf** [comment]
When this pull is merged I will use the following information:
		Version: wf-uicomponents 1.1.4
		Release Ticket: HY-1959

***
*Last updated on Wednesday, April 08 02:43 PM CST*

File: src/awesome_map/EventSynthesizer.js [hunk 1]
```diff
@@ -95,13 +108,11 @@ define(function(require) {
         //---------------------------------------------------------
         // Private properties
         //---------------------------------------------------------
-
-        /**
-         * Cancel mouse wheel events so that browser window does not shift.
-         * @type {boolean}
-         */
```
Comment (tomconnell-wf) [inline src/awesome_map/EventSynthesizer.js]: Duplicate docstring of the configuration parameter.

File: src/awesome_map/EventSynthesizer.js [hunk 2]
```diff
@@ -148,11 +162,6 @@ define(function(require) {
          */
         this._hammer = null;
 
-        /**
-         * The element to watch for events.
-         * @type {HTMLElement}
-         * @private
-         */
```
Comment (tomconnell-wf) [inline src/awesome_map/EventSynthesizer.js]: Duplicate docstring of the configuration parameter.

File: src/awesome_map/EventSynthesizer.js [hunk 3]
```diff
@@ -252,8 +271,7 @@ define(function(require) {
             // The slate is cleaned on release events.
             if (lastGesture) {
                 iterativeGesture = gesture.createIterativeGesture(lastGesture);
-            }
-            else {
```
Comment (tomconnell-wf) [inline src/awesome_map/EventSynthesizer.js]: jscs fixed all these else styles.

File: src/awesome_map/EventSynthesizer.js [hunk 4]
```diff
@@ -170,6 +179,14 @@ define(function(require) {
...
         this._transforming = false;
 
+        /**
+         * Used to be explicit that the event listeners are not using capture. The consistency also
+         * makes sure that we remove the same listener as we add. See
+         * https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener
+         * @type {Boolean}
+         */
+        this._useCapture = false;
+
```
Comment (todbachman-wf) [inline src/awesome_map/EventSynthesizer.js]: This is a nice addition. :+1:

File: src/awesome_map/EventSynthesizer.js [hunk 5]
```diff
@@ -460,6 +483,17 @@ define(function(require) {
...
 
+        _getMsHoldVisualHandler: function() {
+            var self = this;
+
+            return function(event) {
+                if (self._cancelTouchContextMenuEvents) {
+                    // Prevent the Windows "box hint" that appears during a hold.
+                    event.preventDefault();
+                }
+            };
```
Comment (todbachman-wf) [inline src/awesome_map/EventSynthesizer.js]: Maybe this should be (yet another) configuration option that defaults to true. Call it `cancelMsHoldVisual`...

**Comment by todbachman-wf** [comment]
+1. Nice work @tomconnell-wf!

**Comment by robbecker-wf** [comment]
+10 so much nicer
notouchy

**Comment by charliekump-wf** [comment]
QA Resource Approval: +1

- [x] Testing instruction
- [x] Dev +1's
- [x] Dev/QA +10 with detail of what was tested
- [x] Unit test created/updated 
- [x] All unit tests pass 

Merging into master.

**Pull request merged**